### DISCUSSION TITLE
Are HTTP 500 errors due to statement timeouts related to database server performance limitations?

### DISCUSSION THREAD

**Discussion by datsenkoboos** [supabase/supabase#27569] [Questions] [answered]
Hi! We have a table with 15 columns and ~120,000 rows. When calling an RPC function to get distinct column values, we frequently encounter HTTP 500 errors with a "canceling statement due to statement timeout" message.

I would like to understand if this issue is related to server performance limitations. Would upgrading our server with the Compute Add-On help resolve this problem?

The RPC function used to query the table:

```
CREATE OR REPLACE FUNCTION get_distinct_values(table_name text, field text)
RETURNS TABLE(value text)
LANGUAGE plpgsql
AS $$
BEGIN
  RETURN QUERY EXECUTE FORMAT('SELECT DISTINCT %I::text FROM %I', field, table_name);
END;
$$;
```

Thank you

**Comment by GaryAustin1** [comment, marked as answer]
Most likely it would, but you should make sure you have optimized the query and tuned RLS if used on the table.
https://github.com/orgs/supabase/discussions/21133
https://github.com/orgs/supabase/discussions/22449
https://supabase.com/docs/guides/database/postgres/row-level-security#rls-performance-recommendations
Increasing timeouts:  https://supabase.com/docs/guides/database/postgres/configuration#changing-the-default-timeout
